### PR TITLE
fixes gh-1263: ensure that opened widgets are rendered and activated

### PR DESCRIPTION
**Pull request by akosyakov**

### CODE REVIEW

**Comment by AlexTugarev** [comment]
Works great so far with the monaco editor test case.

File: packages/core/src/browser/widget-open-handler.ts [hunk 1]
```diff
@@ -71,22 +72,50 @@ export abstract class WidgetOpenHandler<W extends BaseWidget> implements OpenHan
...
+    }
+    protected onActive(widget: W): Promise<void> {
+        return new Promise(resolve => {
+            const listener = (shell: ApplicationShell, args: FocusTracker.IChangedArgs<Widget>) => {
+                if (args.newValue === widget) {
+                    this.shell.activeChanged.disconnect(listener);
+                    resolve();
+                }
+            };
+            this.shell.activeChanged.connect(listener);
```
Comment (AlexTugarev) [inline packages/core/src/browser/widget-open-handler.ts]: could it happen, that widget is already active and `activeChange` is not emitted?

Comment (AlexTugarev) [reply packages/core/src/browser/widget-open-handler.ts]: I know, on blur of widget, shell won't say this widget is active.

Comment (AlexTugarev) [reply packages/core/src/browser/widget-open-handler.ts]: Maybe it's bike-shedding, and I only can think of a synthetic edge case where some flawed command (maybe triggered via key binding) would try to open same widget as the currently active one.

Comment (akosyakov) [reply packages/core/src/browser/widget-open-handler.ts]: I will add the guard to check whether the widget is already active. It also can be a problem if a  widget does not focus anything but then it does not play nicely with the shell as well.

Comment (AlexTugarev) [reply packages/core/src/browser/widget-open-handler.ts]: thanks!

Comment (akosyakov) [reply packages/core/src/browser/widget-open-handler.ts]: I've added guards

**Pull request merged**